backends/rabbitmq-streams/read.go

```diff
@@ -0,0 +1,106 @@
+package rabbitmqStreams
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/batchcorp/plumber/printer"
+
+	"github.com/batchcorp/plumber/reader"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/pkg/errors"
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/amqp"
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream"
+	"github.com/sirupsen/logrus"
+
+	"github.com/batchcorp/plumber/cli"
+)
+
+func Read(opts *cli.Options, md *desc.MessageDescriptor) error {
+	client, err := NewClient(opts)
+	if err != nil {
+		return err
+	}
+
+	r := &RabbitMQStreams{
+		Client:  client,
+		Options: opts,
+		MsgDesc: md,
+		log:     logrus.WithField("pkg", "rabbitmq-streams/read.go"),
+	}
+
+	return r.Read()
+}
+
+func (r *RabbitMQStreams) Read() error {
+	var count int
+
+	offsetOption, err := r.getOffsetOption()
+	if err != nil {
+		return errors.Wrap(err, "could not read messages")
+	}
+
+	handleMessage := func(consumerContext stream.ConsumerContext, message *amqp.Message) {
+		for _, value := range message.Data {
+			count++
+			data, err := reader.Decode(r.Options, r.MsgDesc, value)
+			if err != nil {
+				r.log.Error(err)
+				continue
+			}
+
+			printer.PrintRabbitMQStreamsResult(r.Options, count, consumerContext, message, data)
+		}
+
+		if !r.Options.ReadFollow {
+			consumerContext.Consumer.CloseHandler <- stream.Event{}
+			consumerContext.Consumer.Close()
+		}
+	}
+
+	defer r.Client.Close()
+
+	consumer, err := r.Client.NewConsumer(r.Options.RabbitMQStreams.Stream,
+		handleMessage,
+		stream.NewConsumerOptions().
+			SetConsumerName(r.Options.RabbitMQStreams.ClientName).
+			SetOffset(offsetOption))
+	if err != nil {
+		return errors.Wrap(err, "unable to start rabbitmq streams consumer")
+	}
+
+	r.log.Infof("Waiting for messages on stream '%s'...", r.Options.RabbitMQStreams.Stream)
+
+	closeCh := consumer.NotifyClose()
+
+	select {
+	case <-closeCh:
```

Comment:
you can receive the close event not only in case of error but also if you `unsubscribe`  [close event](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/main/pkg/stream/listeners.go#L3) has few details you use to determine why a consumer is closed.

check  this  [example](https://gist.github.com/Gsantomaggio/cceafd49af8662f78e9be216c920de60)

Reply:
I guess it would be good to know why it closed :) Added